packages/common-all/src/utils/index.ts

```diff
@@ -1286,6 +1286,93 @@ export class ConfigUtils {
   static getConfigDescription = (conf: string) => {
     return _.get(DENDRON_CONFIG, conf)?.desc;
   };
+
+  /**
+   * Given an config object and an optional array of lodash property path,
+   * omit the properties from the object and flatten it
+   * The result will be a flat array of path-value pairs
+   *
+   * Each pair will contain a path and a value.
+   * The value is either a primitive value, or a stringified array.
+   *
+   * If comparing the array value of a config is unnecessary,
+   * make sure to add it to the omit path.
+   */
+  static flattenConfigObject(opts: { obj: Object; omitPaths?: string[] }) {
+    const { obj, omitPaths } = opts;
+    if (omitPaths && omitPaths.length > 0) {
+      omitPaths.forEach((path) => {
+        _.unset(obj, path);
```

Comment:
isn't this mutating the original config? can we make a copy?

Reply:
right. good catch!